### PR TITLE
[Text Generation][Fix] Update the logic from sampling from a distribution when `deterministic=False`

### PR DESCRIPTION
**Pull request by dbogunowicz**
The function `self.generate_token` expects to receive a single logits array of size `(vocab_size,)` and not `(batch_size, vocab_size)`.

### CODE REVIEW

File: src/deepsparse/transformers/engines/nl_decoder_engine.py [hunk 1]
```diff
@@ -155,7 +156,7 @@ def __call__(
         else:
             logits = out[0]
 
-        token = self.generate_token(logits=logits[:, -1, :])
+        token = self.generate_token(logits=logits[0, -1, :])
```
Comment (bfineran) [inline src/deepsparse/transformers/engines/nl_decoder_engine.py:159]:
```suggestion
        # select batch idx 0, batch is always 1
        token = self.generate_token(logits=logits[0, -1, :])
```

**Review by bfineran — APPROVED** [review]
